src/src/components/PowerTools/GhostEvents.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Real calendar event ID length is around 74 characters, but in order to reduce
+ * used storage, we store only the first 16 characters of the ID.
+ */
+import React from 'react';
+
+import { useSafeStorage } from '../../hooks/useStorage';
+import { listen } from '../../utils/events';
+import { usePref } from '../Preferences/usePref';
+
+const trimmedIdLength = 16;
+const ghostEventStyle = `{
+  opacity: 0.3;
+  pointer-events: none;
+}`;
+
+export function GhostEvents(): null {
+  const [ghostEvents = [], setGhostEvents] = useSafeStorage('ghostEvents', []);
+  const styleRef = React.useRef<HTMLStyleElement | undefined>(undefined);
+  const [ghostEventShortcut] = usePref('features', 'ghostEventShortcut');
+
+  // Set styles
+  React.useEffect(() => {
+    if (styleRef.current === undefined) {
+      styleRef.current = document.createElement('style');
+      document.head.append(styleRef.current);
+    }
+
+    // This allows for a way to temporary disable event ghosting
+    const isDisabled = ghostEventShortcut === 'none';
+
+    styleRef.current.textContent =
+      isDisabled || ghostEvents.length === 0
+        ? ''
+        : `${ghostEvents
+            .map((id) => `[data-eventid*="${id}"]`)
+            .join(',')}${ghostEventStyle}`;
+  }, [ghostEvents, ghostEventShortcut]);
+
+  // Listen for key press
+  React.useEffect(
+    () =>
+      ghostEventShortcut === 'none'
+        ? undefined
+        : listen(
+            document.body,
+            'click',
+            ({ shiftKey, ctrlKey, metaKey, target }) => {
+              const keys = {
+                shift: shiftKey,
+                ctrl: ctrlKey,
+                cmd: metaKey,
+              };
+              if (!keys[ghostEventShortcut]) return;
+              const isSingleKey =
+                [shiftKey, ctrlKey, metaKey].filter(Boolean).length === 1;
+              if (!isSingleKey) return;
+              const eventElement = (target as HTMLElement).closest(
+                '[data-eventid]'
+              );
+              if (eventElement === null) return;
+              const eventId = eventElement.getAttribute('data-eventid') ?? '';
+              if (eventId.length === 0) return;
+              const trimmed = eventId.slice(0, trimmedIdLength);
```

Comment:
Event ID consists of A-Z, a-z and 0-9 (total 62 characters)
There are `62**16` possible IDs of length 16
That's `4.7672402e+28` - a really really big number.
I won't be worried about collisions. Similar to the `hash collision` problem.

Regarding why first 16, no particular reason. You are probably right though that using last 16 would be better (I have seen a disproportionate amount of IDs starting with `N`)

Reply:
Though, it sounds like third-party apps can pick arbitrary IDs - not good as they can all have similar prefixes or endings

![Screenshot 2022-10-31 at 14 03 56](https://user-images.githubusercontent.com/40512816/199089122-0673a8fb-3a00-46aa-a4ce-cba4d73929d6.png)

([source](https://developers.google.com/calendar/api/v3/reference/events#:~:text=Opaque%20identifier%20of%20the%20event))

The current solution of ghosting by eventID is just a first solution. Probably the best implementation would be to use the event name, as then if you have multiple events by the same name, you don't have to manually ghost each of them.

I begun with the eventId solution because it allows for marking events as ghosts using a CSS style (because events have an html `data-eventid` attribute). For a more sophisticated solution, we would have to scan the page for rendered events.